flow/project.py

```diff
@@ -1818,43 +2021,59 @@ def _fetch_status(self, jobs, err, ignore_errors, status_parallelization='thread
         # to do proper deprecation, it is not required for now.
 
         # Update the project's status cache
-        self._fetch_scheduler_status(jobs, err, ignore_errors)
-        # Get status dict for all selected jobs
-
-        def _print_progress(x):
-            print("Updating status: ", end='', file=err)
-            err.flush()
-            n = max(1, int(len(jobs) / 10))
-            for i, _ in enumerate(x):
-                if (i % n) == 0:
-                    print('.', end='', file=err)
-                    err.flush()
-                yield _
-
+        self._fetch_scheduler_status(aggregates, err, ignore_errors)
+
+        def _print_status(iterable, fetch_status, description):
+            t = time.time()
+            num_itr = len(iterable)
+            results = []
+            for i, itr in enumerate(iterable):
+                results.append(fetch_status(itr))
+                # The status interval 0.2 seconds is used since we expect the
+                # status for an aggregate to be fetched within that interval
+                if time.time() - t > 0.2:
+                    print(f'{description}: {i+1}/{num_itr}', end='\r', file=err)
+                    t = time.time()
+            # Always print the completed progressbar.
+            print(f'{description}: {i+1}/{num_itr}', file=err)
+            return results
+
+        # Get status dict for all selected aggregates
         try:
             cached_status = self.document['_status']._as_dict()
         except KeyError:
             cached_status = dict()
 
-        _get_job_status = functools.partial(self.get_job_status,
-                                            ignore_errors=ignore_errors,
-                                            cached_status=cached_status)
+        _get_job_labels = functools.partial(self._get_job_labels,
+                                            ignore_errors=ignore_errors)
+
+        _get_group_status = functools.partial(self._get_group_status,
+                                              ignore_errors=ignore_errors,
+                                              cached_status=cached_status)
+
+        singleton_groups = [op for op in self.operations]
```

Comment:
We previously showed operation based status and not a group based on. Meaning the status for a job operation pair will be shown and not a job group pair.
But now since aggregates are associated with groups, I thought that it'd be logical to name the variable as `singleton_groups` and stored all the operation names in it. 

I think this variable is poorly named, should I change this to `operation_as_group`?

Reply:
Let's rename it. A lot of code is inconsistent with names (groups vs. operations), so I think the choice between `all_groups` and `all_operations` is not important to me. They're actually groups, but the data is from `self.operations`. 🤷‍♂️
```suggestion
        all_operations = list(self.operations.keys())
```

Reply:
I renamed this to `operation_names` to clarify that behavior. I think this can be resolved.